packages/server/src/unstable-core-do-not-import/stream/jsonl.ts

```diff
@@ -117,101 +112,87 @@ class MaxDepthError extends Error {
   }
 }
 
-function createBatchStreamProducer(opts: ProducerOptions) {
+async function* createBatchStreamProducer(
+  opts: ProducerOptions,
+): AsyncIterable<Head | ChunkData, void> {
   const { data } = opts;
   let counter = 0 as ChunkIndex;
   const placeholder = 0 as PlaceholderValue;
 
-  const stream = createReadableStream<ChunkData>();
-  const pending = withRefCount(new Set<ChunkIndex>(), () => {
-    if (!stream.cancelled()) {
-      stream.controller.close();
-    }
-  });
+  const set = new Set<{
```

Comment:
Rename `const set` to e.g. `const queue`, as we do not use any features that `Set` is known for ⇒ confusing to read. Also, add a comment on why we use `Set` (simple and fast `delete` API)